strr-web/components/bcros/form-section/principal-residence/Form.vue

```diff
@@ -224,7 +225,12 @@ const uploadFile = (file: FileList) => {
     fileError.value = tPrincipalResidence('fileTypeError')
   } else {
     fileError.value = null
-    formState.supportingDocuments.push(file[0])
+    const existingIndex = formState.supportingDocuments.findIndex(doc => doc.name === newFile.name)
+    if (existingIndex !== -1) {
+      formState.supportingDocuments.splice(existingIndex, 1, newFile)
```

Comment:
We can simplify here and just set the `formState.supportingDocuments[existingIndex] = newFile`